### DISCUSSION TITLE
Please explain why these conditional sections are guarded by __java__

### DISCUSSION THREAD

**Discussion by vsajip** [congo-cc/congo-parser-generator#108] [Q&A] [answered]
This line:
https://github.com/congo-cc/congo-parser-generator/blob/6e61316e5613866ceaac703f4024a08790a05478/examples/csharp/CSharp.ccc#L2361
and several other identical lines were added (according to `git blame`) in a commit that says it was addressing #22 and #23, but they appear unrelated - can you explain why the conditionals are there, please, @revusky ?

**Comment by revusky** [comment]
You're right. The issues 22 and 23 are totally unrelated to that line in the CSharp grammar. I guess the change was there and it ended up being committed with the other unrelated changes.

As for why that line is conditioned so that it is ignored for non-java languages, I guess the reason is that, for some reason, that production, `InterpolatedRawString` doesn't work except in Java. But I can't remember why. Or maybe it does work now. I honestly can't remember much more at the moment.

**Comment by revusky** [comment, marked as answer]
Actually, looking at this a bit more, I guess the key is that the non-Java translators were not handling the code injection [here](https://github.com/congo-cc/congo-parser-generator/blob/6e61316e5613866ceaac703f4024a08790a05478/examples/csharp/CSharpLexer.ccc#L362-L412). I probably made a mental note to get back to this but never did, I guess,

**Comment by vsajip** [comment]
OK, that's helpful. I can look at why those injections are problematic and see if I can remove the conditionals.

**Comment by vsajip** [comment]
Perhaps, related, is this:
https://github.com/congo-cc/congo-parser-generator/blob/6e61316e5613866ceaac703f4024a08790a05478/src/grammars/CSharpInternal.ccc#L8
also a holdover which can now be removed? It appears nowhere else in the code.
